File: LeapYear.py
def leap_Year(year):
    """Return True if year is a leap year, else False, and print the result."""
    # Leap year if divisible by 400, or divisible by 4 but not by 100
    if (year % 400 == 0) or ((year % 4 == 0) and (year % 100 != 0)):
        print(year, "is a leap year")
        return True
    else:
        print(year, "is not a leap year")
        return False
        

year = int(input("Enter a year: "))
leap_Year(year)

File: Leap_Year.java
import java.util.*;

public class Leap_Year {

    public static boolean Leap_Year(int year) {
        if ((year % 400 == 0) || ((year % 4 == 0) && (year % 100 != 0))) {
            System.out.println(year + " is a leap year");
            return true;
        } else {
            System.out.println(year + " is not a leap year");
            return false;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter a year: ");
        int year = sc.nextInt();
        Leap_Year(year);
        sc.close();
    }
}